### DISCUSSION TITLE
Support for wildcard workspace members?

### DISCUSSION THREAD

**Discussion by mightyiam** [juspay/rust-flake#28] [Q&A]
```toml
[workspace]
members = ["crates/*"]
```

I got an error for this

**Comment by srid** [comment]
Does Nix have a library function to iterate paths using glob?

**Reply by mightyiam** [reply]
[This](https://github.com/pdtpartners/globset) doesn't count for one, does it?

**Reply by srid** [reply]
Upstream is better so this flake doesn't have to take a whole another flake as dependency just for this little convenience.

**Reply by srid** [reply]
https://github.com/juspay/rust-flake/blob/07c591ee5edf846dc70ef9c5f8af8333ced3bdb2/nix/modules/default-crates.nix#L45

Note that: you *can* define the `rust-project.crates` option manually in Nix, superceding the workspace members from cargo toml.

**Reply by srid** [reply]
> Upstream is better so this flake doesn't have to take a whole another flake as dependency just for this little convenience.

That said, we can also probably `builtins.fetch` that flake lazily if `*` is used and then do the magic.

**Reply by mightyiam** [reply]
Also to consider is that not all glob implementations are the same. And we're interested in Cargo's particular one.
